### DISCUSSION TITLE
Noob question on extracting position size for backtesting.py

### DISCUSSION THREAD

**Discussion by kiann00** [kernc/backtesting.py#858] [Q&A]
Hi noob on python and backtesting.py there,

I have a relatively simple customized technical indicator which produces/produced a signal on every time-slice. What I want to do is to have a maximum position size, which I tried to incorporate in the 'Strategy' class as per below :
1. Somehow, the code below doesn't seem to max out the position (Noticed it doesn't seem to be correct though, my portfolio seems to allow position of more than 3 positions.)
2. Worse, for some reason, the backtesting.py (Strategy) doesn't seem to be picking up the buy/sell signals.
    - my logic is to buy when signal is above 1, and short when signal is smaller than -1 (these magnitudes will be changed)
    - but somehow, it is doing only 1 trade across the whole time-periods.

Help!!! @kernc  / or any experts on this module here?  🙏🙏🙏

```
class SignalStrategy(Strategy):
    def init(self):
        pass
    ...
    def next(self):     
        current_signal = self.data.signal[-1]
        # if buy signal and position size is less than max position
        if current_signal >= 1 & (abs(position.size) <= 3):
            if not self.position:
                self.buy()
        # if SELL signal and position size is less than max position
        elif (current_signal <= -1) & (abs(position.size) <= 3) :
            if not self.position:
                self.sell()
```

Below are my signal and price data fed into the code : bt = Backtest(tmp2_, SignalStrategy, cash=10_000)

tmp2_ dataframe below
> 	Open_5min	High_5min	Low_5min	Close_5min	target	signal
> 2022-10-06 09:30:00	375.829987	377.049988	375.720001	376.859985	-0.001672	1
> 2022-10-06 09:35:00	376.859985	377.619995	376.160004	377.600708	-0.006596	-3
> 2022-10-06 09:40:00	377.609985	378.720001	377.470001	378.029999	-0.008729	-3
> 2022-10-06 09:45:00	378.029999	378.179993	376.859985	377.165009	-0.007782	-2
> 2022-10-06 09:50:00	377.140015	378.168488	376.950012	377.760010	-0.008286	-4
> 2022-10-06 09:55:00	377.731903	377.760010	376.000000	376.230011	-0.004678	-1
> 2022-10-06 10:00:00	376.230011	376.260010	374.709991	375.109985	-0.003865	0
> 2022-10-06 10:05:00	375.109985	375.119995	374.190002	374.730011	-0.003202	1
> 2022-10-06 10:10:00	374.730011	374.739990	373.779999	374.230011	-0.000053	2
> 2022-10-06 10:15:00	374.209991	375.230011	374.200012	374.630005	0.000107	2
> 2022-10-06 10:20:00	374.329987	374.839996	374.109985	374.469910	0.003459	2
> 2022-10-06 10:25:00	374.459991	374.820007	373.609985	373.660004	0.005807	3
> 2022-10-06 10:30:00	373.660095	374.309998	373.440002	373.529999	0.007737	4
> 2022-10-06 10:35:00	373.529999	374.540009	373.410004	374.209991	0.002779	4
> 2022-10-06 10:40:00	374.204987	374.959991	374.160095	374.670013	0.002375	3
> 2022-10-06 10:45:00	374.679993	375.809998	374.350098	375.765015	-0.002014	2
> 2022-10-06 10:50:00	375.750000	375.920013	375.399994	375.829987	0.001330	2
> 2022-10-06 10:55:00	375.858002	376.489990	375.450012	376.420013	0.000505	1
> 2022-10-06 11:00:00	376.429993	376.559998	375.089996	375.250000	0.003358	3
> 2022-10-06 11:05:00	375.250000	376.052002	375.250000	375.559998	0.001091	2

FYI, the data is SPY in which the price is about 385.0

![image](https://user-images.githubusercontent.com/26670633/210246007-cdb943a8-e3d3-4ff1-bb46-eaecdbfe0d1e.png)

**Comment by iimka** [comment]
position size =-26
abs(position.size) =26 > 3
cannot trigger any conditions
